### PR TITLE
Add create new project

### PR DESCRIPTION
**Pull request by rainy789**
## Motivation
Create a new project.

## Modification
- Add create new project rtk mutation.
- Add NewProject component.
- Redirect to the newly created project page on success.
- Clean up: fix multiple lint warnings.

## Result

Screen Shot 2023-01-11 at 07 35 36

Screen Shot 2023-01-11 at 07 36 44

### CODE REVIEW

**Comment by ikhoon** [comment]
It is not related to this PR, but I found that there is an API mismatch in the project meta API.
https://github.com/line/centraldogma/blob/eb010e790d18fdc98fa1d08517829eb2f86f96f2/webapp/src/dogma/features/api/apiSlice.ts#L67
The server-side handles only non-trailing slash API.
https://github.com/line/centraldogma/blob/e46e977e8c2836047e44f7389b00da41412a44a5/server/src/main/java/com/linecorp/centraldogma/server/internal/api/ProjectServiceV1.java#L105-L113
If it is difficult to handle a non-trailing slash because of the limitation of NextJS routing, I'm OK to make `getProjectMetadata()` handle `/projects/{projectName}/` as well.

**Review by ikhoon — APPROVED** [review]
Everything looks good!

**Comment by ikhoon** [comment]
Let me fix https://github.com/line/armeria/issues/4542 quickly and then set the following options to `false`. https://github.com/line/centraldogma/blob/d4c6397baf7367d81677c90cdd5537dbf15ebe62/webapp/next.config.js#L6

**Comment by rainy789** [comment]
> The server-side handles only non-trailing slash API.
I will remove the trailing slash from the API endpoints then. Will handle it in a separate PR.

File: webapp/src/dogma/common/components/Navbar.tsx [hunk 1]
```diff
@@ -164,11 +165,14 @@ export const Navbar = () => {
             }}
           />
         </Box>
-        <Flex alignItems="center">
-          <Button variant="solid" colorScheme="teal" size="sm" mr={4} leftIcon={<AddIcon />}>
-            New Project
-          </Button>
-          <Button onClick={toggleColorMode}>{colorMode === 'light' ? <MoonIcon /> : <SunIcon />}</Button>
+        <Flex alignItems="center" gap={2}>
+          <NewProject />
```
[the remark this answers is not in the document]
Comment (rainy789) [reply webapp/src/dogma/common/components/Navbar.tsx:169]: Let's explore this when we're adding `NewRepository` functionality.

**Comment by rainy789** [comment]
> Let me fix [line/armeria#4542](https://github.com/line/armeria/issues/4542) quickly and then set the following options to `false`.
> 
> https://github.com/line/centraldogma/blob/d4c6397baf7367d81677c90cdd5537dbf15ebe62/webapp/next.config.js#L6

Isn't this only related to pages? It probably doesn't affect API.

**Comment by ikhoon** [comment]
> Isn't this only related to pages? It probably doesn't affect API.

If so, we can create a mock API for `/api/v1/projects/{projectName}` under `/pages/api/v1/...`?

**Comment by rainy789** [comment]
> > Isn't this only related to pages? It probably doesn't affect API.
> 
> If so, we can create a mock API for `/api/v1/projects/{projectName}` under `/pages/api/v1/...`?

I removed the `/` from the endpoint and it seems to work.
Screen Shot 2023-01-11 at 14 21 10


Screen Shot 2023-01-11 at 14 19 54


Will include this change in an upcoming PR.

**Comment by ikhoon** [comment]
> I tried removing the / from the endpoint and it seems to work.

Sounds good. Thanks for checking.

File: webapp/src/dogma/common/components/Navbar.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (rainy789) [reply webapp/src/dogma/common/components/Navbar.tsx:169]: `NewRepository` is now available on the repository section. I think it's better this way. https://github.com/line/centraldogma/pull/788

**Pull request merged**